exec/java-exec/src/test/resources/functions/testBitTwiddlers.json

```diff
@@ -0,0 +1,34 @@
+{
```

Comment:
Suggestion: this form of test was used very early on in Drill before the full SQL stack worked. Consider writing a SQL-based test.

Since we are trying to support all types, I wonder if we should have a test which generates the needed SQL. Something like:

```sql
SELECT CAST(A AS SMALLINT) XOR CAST(B AS SMALLINT)
FROM (VALUES ...)
```

Then generate the SQL for each supported data type and verify the results.

The `CAST` is suggested because there is no good way in Drill to create a data file that has data of every types (since Drill has no way to specify a schema with data types.) I suppose you could create a Parquet file with the required types, but that seems overkill.

Reply:
Thanks for the review @paul-rogers, I'll certain allow more time for reviews to arrive, in future.  The naming of these functions put me into a bit of a spin, to the point where I couldn't formulate very good questions about naming when I asked on the Slack channel.

1. I started out trying for names of `AND`, `OR` and `XOR` but the first two seem to be taken elsewhere Drill and would not work, while the last is fine and indeed was already synonymous with `^` before this patch (but only working for 32-bit arguments).
2. Investigating `AND` and `OR`, I found `booleanAnd` and `booleanOr` in FunctionNames.java and then also logic in other code using these names to decide if something is a boolean expression, which bitwise logical operators are not.  I decided at that point that I should not overload function names that were intended to be boolean, and abandoned `AND` and `OR`.
3. I considered `BIT_AND`, `BIT_OR` and `BIT_XOR` but already these belong to the aggregate bitwise functions and do not allow overloading for the scalar case (I got runtime errors when I tried).
4. I considered `INT_AND`, `INT_OR` and `INT_XOR` and indeed these did work in testing.
5. Somehow I uncovered that `&`, `|`, `^` are _already_ working scalar bitwise functions in Drill, in spite of my being unable to find any implementation of them in the code.  Looking at physical query plans did not give me any clues and I told myself that some "magic" is happening upstream, possibly code gen by Calcite?
6. Since Drill already has the names in (5) for bitwise functions, even if they're made of special characters and nasty, I stuck with them because it's consistent with what's already there.

I'd be grateful for any suggestion of where to take this in a follow-up commit...